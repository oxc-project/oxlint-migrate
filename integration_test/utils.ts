import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
