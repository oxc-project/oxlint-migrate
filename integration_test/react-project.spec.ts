import { expect, test } from 'vitest';
// @ts-ignore
import puppeteer_test from './projects/react-project.eslint.config.mjs';
import { getSnapshotResult, getSnapShotMergeResult } from './utils.js';

test('react-project', async () => {
  const result = await getSnapshotResult(puppeteer_test);
  expect(result).toMatchSnapshot('react-project');
});

test('react-project merge', async () => {
  const result = await getSnapShotMergeResult(puppeteer_test, {
    categories: {
      correctness: 'error',
      perf: 'error',
    },
  });
  expect(result).toMatchSnapshot('react-project--merge');
});
