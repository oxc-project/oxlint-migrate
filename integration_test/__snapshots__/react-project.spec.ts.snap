// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react-project > react-project 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [
      "react",
      "react-perf",
    ],
    "rules": {
      "react-hooks/exhaustive-deps": "warn",
      "react-hooks/rules-of-hooks": "error",
      "react-perf/jsx-no-new-array-as-prop": "warn",
      "react-perf/jsx-no-new-function-as-prop": "warn",
      "react-perf/jsx-no-new-object-as-prop": "warn",
      "react/jsx-key": "warn",
      "react/jsx-no-comment-textnodes": "warn",
      "react/jsx-no-duplicate-props": "warn",
      "react/jsx-no-target-blank": "warn",
      "react/jsx-no-undef": "warn",
      "react/no-children-prop": "warn",
      "react/no-danger-with-children": "warn",
      "react/no-direct-mutation-state": "warn",
      "react/no-find-dom-node": "warn",
      "react/no-is-mounted": "warn",
      "react/no-render-return-value": "warn",
      "react/no-string-refs": "warn",
      "react/no-unescaped-entities": "warn",
      "react/no-unknown-property": "warn",
    },
  },
  "warnings": [
    "unsupported rule: react/display-name",
    "unsupported rule: react/jsx-uses-react",
    "unsupported rule: react/jsx-uses-vars",
    "unsupported rule: react/no-deprecated",
    "unsupported rule: react/prop-types",
    "unsupported rule, but in development: react/require-render-return",
    "unsupported rule: react-hooks/static-components",
    "unsupported rule: react-hooks/use-memo",
    "unsupported rule: react-hooks/void-use-memo",
    "unsupported rule: react-hooks/component-hook-factories",
    "unsupported rule: react-hooks/preserve-manual-memoization",
    "unsupported rule: react-hooks/incompatible-library",
    "unsupported rule: react-hooks/immutability",
    "unsupported rule: react-hooks/globals",
    "unsupported rule: react-hooks/refs",
    "unsupported rule: react-hooks/set-state-in-effect",
    "unsupported rule: react-hooks/error-boundaries",
    "unsupported rule: react-hooks/purity",
    "unsupported rule: react-hooks/set-state-in-render",
    "unsupported rule: react-hooks/unsupported-syntax",
    "unsupported rule: react-hooks/config",
    "unsupported rule: react-hooks/gating",
  ],
}
`;

exports[`react-project --js-plugins > react-project--js-plugins 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [
      "react",
      "react-perf",
    ],
    "rules": {
      "react-hooks/exhaustive-deps": "warn",
      "react-hooks/rules-of-hooks": "error",
      "react-perf/jsx-no-new-array-as-prop": "warn",
      "react-perf/jsx-no-new-function-as-prop": "warn",
      "react-perf/jsx-no-new-object-as-prop": "warn",
      "react/jsx-key": "warn",
      "react/jsx-no-comment-textnodes": "warn",
      "react/jsx-no-duplicate-props": "warn",
      "react/jsx-no-target-blank": "warn",
      "react/jsx-no-undef": "warn",
      "react/no-children-prop": "warn",
      "react/no-danger-with-children": "warn",
      "react/no-direct-mutation-state": "warn",
      "react/no-find-dom-node": "warn",
      "react/no-is-mounted": "warn",
      "react/no-render-return-value": "warn",
      "react/no-string-refs": "warn",
      "react/no-unescaped-entities": "warn",
      "react/no-unknown-property": "warn",
    },
  },
  "warnings": [
    "unsupported rule: react/display-name",
    "unsupported rule: react/jsx-uses-react",
    "unsupported rule: react/jsx-uses-vars",
    "unsupported rule: react/no-deprecated",
    "unsupported rule: react/prop-types",
    "unsupported rule, but in development: react/require-render-return",
    "unsupported rule: react-hooks/static-components",
    "unsupported rule: react-hooks/use-memo",
    "unsupported rule: react-hooks/void-use-memo",
    "unsupported rule: react-hooks/component-hook-factories",
    "unsupported rule: react-hooks/preserve-manual-memoization",
    "unsupported rule: react-hooks/incompatible-library",
    "unsupported rule: react-hooks/immutability",
    "unsupported rule: react-hooks/globals",
    "unsupported rule: react-hooks/refs",
    "unsupported rule: react-hooks/set-state-in-effect",
    "unsupported rule: react-hooks/error-boundaries",
    "unsupported rule: react-hooks/purity",
    "unsupported rule: react-hooks/set-state-in-render",
    "unsupported rule: react-hooks/unsupported-syntax",
    "unsupported rule: react-hooks/config",
    "unsupported rule: react-hooks/gating",
  ],
}
`;

exports[`react-project --type-aware > react-project--type-aware 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [
      "react",
      "react-perf",
    ],
    "rules": {
      "react-hooks/exhaustive-deps": "warn",
      "react-hooks/rules-of-hooks": "error",
      "react-perf/jsx-no-new-array-as-prop": "warn",
      "react-perf/jsx-no-new-function-as-prop": "warn",
      "react-perf/jsx-no-new-object-as-prop": "warn",
      "react/jsx-key": "warn",
      "react/jsx-no-comment-textnodes": "warn",
      "react/jsx-no-duplicate-props": "warn",
      "react/jsx-no-target-blank": "warn",
      "react/jsx-no-undef": "warn",
      "react/no-children-prop": "warn",
      "react/no-danger-with-children": "warn",
      "react/no-direct-mutation-state": "warn",
      "react/no-find-dom-node": "warn",
      "react/no-is-mounted": "warn",
      "react/no-render-return-value": "warn",
      "react/no-string-refs": "warn",
      "react/no-unescaped-entities": "warn",
      "react/no-unknown-property": "warn",
    },
  },
  "warnings": [
    "unsupported rule: react/display-name",
    "unsupported rule: react/jsx-uses-react",
    "unsupported rule: react/jsx-uses-vars",
    "unsupported rule: react/no-deprecated",
    "unsupported rule: react/prop-types",
    "unsupported rule, but in development: react/require-render-return",
    "unsupported rule: react-hooks/static-components",
    "unsupported rule: react-hooks/use-memo",
    "unsupported rule: react-hooks/void-use-memo",
    "unsupported rule: react-hooks/component-hook-factories",
    "unsupported rule: react-hooks/preserve-manual-memoization",
    "unsupported rule: react-hooks/incompatible-library",
    "unsupported rule: react-hooks/immutability",
    "unsupported rule: react-hooks/globals",
    "unsupported rule: react-hooks/refs",
    "unsupported rule: react-hooks/set-state-in-effect",
    "unsupported rule: react-hooks/error-boundaries",
    "unsupported rule: react-hooks/purity",
    "unsupported rule: react-hooks/set-state-in-render",
    "unsupported rule: react-hooks/unsupported-syntax",
    "unsupported rule: react-hooks/config",
    "unsupported rule: react-hooks/gating",
  ],
}
`;

exports[`react-project merge > react-project--merge 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "error",
      "perf": "error",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [
      "react",
      "react-perf",
    ],
    "rules": {
      "react-hooks/exhaustive-deps": "warn",
      "react-hooks/rules-of-hooks": "error",
      "react-perf/jsx-no-new-array-as-prop": "warn",
      "react-perf/jsx-no-new-function-as-prop": "warn",
      "react-perf/jsx-no-new-object-as-prop": "warn",
      "react/jsx-key": "warn",
      "react/jsx-no-comment-textnodes": "warn",
      "react/jsx-no-duplicate-props": "warn",
      "react/jsx-no-target-blank": "warn",
      "react/jsx-no-undef": "warn",
      "react/no-children-prop": "warn",
      "react/no-danger-with-children": "warn",
      "react/no-direct-mutation-state": "warn",
      "react/no-find-dom-node": "warn",
      "react/no-is-mounted": "warn",
      "react/no-render-return-value": "warn",
      "react/no-string-refs": "warn",
      "react/no-unescaped-entities": "warn",
      "react/no-unknown-property": "warn",
      "react/react-in-jsx-scope": "warn",
    },
  },
  "warnings": [
    "unsupported rule: react/display-name",
    "unsupported rule: react/jsx-uses-react",
    "unsupported rule: react/jsx-uses-vars",
    "unsupported rule: react/no-deprecated",
    "unsupported rule: react/prop-types",
    "unsupported rule, but in development: react/require-render-return",
    "unsupported rule: react-hooks/static-components",
    "unsupported rule: react-hooks/use-memo",
    "unsupported rule: react-hooks/void-use-memo",
    "unsupported rule: react-hooks/component-hook-factories",
    "unsupported rule: react-hooks/preserve-manual-memoization",
    "unsupported rule: react-hooks/incompatible-library",
    "unsupported rule: react-hooks/immutability",
    "unsupported rule: react-hooks/globals",
    "unsupported rule: react-hooks/refs",
    "unsupported rule: react-hooks/set-state-in-effect",
    "unsupported rule: react-hooks/error-boundaries",
    "unsupported rule: react-hooks/purity",
    "unsupported rule: react-hooks/set-state-in-render",
    "unsupported rule: react-hooks/unsupported-syntax",
    "unsupported rule: react-hooks/config",
    "unsupported rule: react-hooks/gating",
  ],
}
`;
